xdsl/builder.py

```diff
@@ -258,6 +266,40 @@ def assert_implicit():
             )
 
 
+class SSAValueNameSetter:
+    """
+    Helper structure for setting name hints for inserted ops.
+    """
+
+    _name: str
+    """
+    Valid name for SSAValue objects.
+    """
+
+    def __init__(self, name: str) -> None:
+        self._name = SSAValue.extract_valid_name(name)
+
+    @property
+    def name_hint(self) -> str:
+        return self._name
+
+    @name_hint.setter
+    def name_hint(self, name: str):
+        self._name = SSAValue.extract_valid_name(name)
+
+    def register(self, builder_listener: BuilderListener):
+        """
+        Adds a handler to the builder_listener instance to set the name hint of all
+        results of inserted operations, as long as the result name is not None.
```

Comment:
```suggestion
        results of inserted operations, as long as the result name is None.
```
I think?

Reply:
I'll reword this completely so it's less ambiguous